access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java

```diff
@@ -290,12 +290,20 @@ public CompletionStage<AccessGrant> issue(final URI type, final URI agent, final
         Objects.requireNonNull(type, "Access Grant type may not be null!");
         Objects.requireNonNull(resources, "Resources may not be null!");
         Objects.requireNonNull(modes, "Access modes may not be null!");
+        final Set<URI> uriPurposes = new HashSet<>();
+        for (final String p : purposes) {
+            try {
+                uriPurposes.add(URI.create(p));
+            } catch (final IllegalArgumentException ex) {
+                LOGGER.debug("Ignoring non-URI purpose: {}", ex.getMessage());
```

Comment:
This sounds like a serious change in behaviour.

I think you should definitely document it. Maybe an `@ImplNote`?

Reply:
Actually this all new functionality, no? So not a change.

But definitely document.

Reply:
Addressed in https://github.com/inrupt/solid-client-java/pull/491/commits/3b79f521c2335cd4078ed54e706b5e9b7a4a81fe